cmd/fetch-tsa-certs/fetch_tsa_certs.go

```diff
@@ -145,6 +148,49 @@ func fetchCACertificate(ctx context.Context, parent, kmsKey string, client *priv
 		}
 		parsedCerts = append(parsedCerts, certs[0])
 	}
+	intermediate := parsedCerts[0]
+
+	// generate leaf certificate
+	leafKMSSigner, err := kms.Get(ctx, leafKMSKey, crypto.SHA256)
+	if err != nil {
+		return nil, err
+	}
+	leafSigner, _, err := leafKMSSigner.CryptoSigner(ctx, func(err error) {})
```

Comment:
do you need this? why no pass in `leafKMSSigner.PublicKey()` into the `x509.CreateCertificate` below?

Reply:
good call, updated!